homework/unique_ptr/unique_ptr.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include "NullPtrException.hpp"
+
+namespace cs {
+
+template <typename T>
+class unique_ptr {
+public:
+    unique_ptr(T* ptr);
+    unique_ptr() = delete;
+    unique_ptr(unique_ptr&& previousOwner);
+    unique_ptr(const unique_ptr&) = delete;
+    ~unique_ptr();
+
+    T* release();
+    const T* get() const;
+    void reset(T* newPtr);
```

Comment:
```suggestion
    void reset(T* newPtr = nullptr);
```
Default parameter would be useful

Reply:
I've done it differently now